tests/unittest/auto_deploy/multigpu/smoke/test_ad_allreduce_strategies.py

```
@@ -0,0 +1,310 @@
+import yaml
+from _model_test_utils import get_small_model_config
+from click.testing import CliRunner
+from utils.cpp_paths import llm_root  # noqa: F401
+
+from tensorrt_llm.commands.bench import main
+
+# needed since LLM API uses MPI executor pool internally for TP>1, which leaks a thread on shutdown
+pytestmark = pytest.mark.threadleak(enabled=False)
+
+
+class TimeoutError(Exception):
```

Ruff (0.15.21)

[error] Variable `TimeoutError` is shadowing a Python builtin

(A001)